cmd/gcp-controller-manager/app/node_annotater.go

```diff
@@ -207,3 +248,40 @@ func parseNodeURL(nodeURL string) (project, zone, instance string, err error) {
 	instance = parts[2]
 	return
 }
+
+// TODO: move this to instance.Labels. This is gross.
+func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
+	const labelsKey = "kube-labels"
+
+	if instance.Metadata == nil {
+		return nil, fmt.Errorf("instance %q has no labels")
```

Comment:
Return an empty map in this case? Or is it impossible for instance to have no metadata?
If it should be an error, `s/labels/metadata/`

Reply:
Wonder if we need to worry the case when we unintentionally return an empty map (e.g. metadata got wiped out somehow) it might result in all ownedKubeLabels being removed from nodes, which could have unexpected impact.